packages/core/src/zipHardhat.ts

```diff
@@ -0,0 +1,160 @@
+import JSZip from "jszip";
+import type { GenericOptions } from "./build-generic";
+import type { Contract } from "./contract";
+import type { ERC1155Options } from "./erc1155";
+import { printContract } from "./print";
+
+const hardhatConfig = (upgradeable: boolean) => `\
+import { HardhatUserConfig } from "hardhat/config";
+import "@nomicfoundation/hardhat-toolbox";${upgradeable ? `
+import "@openzeppelin/hardhat-upgrades";` : ''}
```

Comment:
I prefer the extra newline. :slightly_smiling_face: We can also use `formatLines` to simplify some of this (see comment on this later).

```suggestion
import "@nomicfoundation/hardhat-toolbox";
${upgradeable ? `import "@openzeppelin/hardhat-upgrades";` : ''}
```